bucket4j-spring-boot-starter/src/main/java/com/giffing/bucket4j/spring/boot/starter/url/PatternUrlMapper.java

```diff
@@ -6,14 +6,14 @@
 
 import static org.springframework.http.server.PathContainer.parsePath;
 
+@SuppressWarnings("unused")
 public class PatternUrlMapper
         implements UrlMapper {
 
-    private final PathPatternParser pathPatternParser =
-            new PathPatternParser();
-
     @Override
     public UrlMatcher getMatcher(String pattern) {
+        var pathPatternParser =
```

Comment:
Is it really needed? The parse implementation creates a new InternalPathPatternParser because the PathPatternParser is not Thread save. It seems that the InternalPathPatternParser only uses the configuration from the PathPatternParser. So a new instance is created because the PathPatternParser is not Thread-Save but because of that we don't need a new instance every time. Maybe I'm wrong...